crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1,3 +1,6 @@
+// Temporary workaround for impl_reflect!(Option/Result false-positive
+#![allow(unused_qualifications)]
+
```

Comment:
Enabling the lint marks the following as to be improved but applying the suggestion (like `--fix` does) doesn't work. Also, I don't know how to ignore the lint in the `proc_macro` without `#![allow` for the whole module.

https://github.com/bevyengine/bevy/blob/75738ed80f3a7baa3eba9d68629ef9b4785cb4c7/crates/bevy_reflect/src/impls/std.rs#L1468-L1482

I neither don't fully understand the actual root cause of the error (as I don't really know what the `proc_macro` is doing here, not familiar with the code) nor know where to report this false-positive correctly to the Rust devs. As it's not a clippy lint not the clippy repo. Maybe directly in the rust repo as it's a false-positive there?

How should we proceed with this?

Reply:
I'm okay proceeding with this workaround, but we should open an issue for it. Others may know better.

Reply:
Yes, I agree, there should be an issue, and it should be linked in the source next to the workaround.

For opening an issue there should be a somewhat minimal reproducible example. And I seem to understand too little to reduce this issue in some ways.
At the current state I don't feel confident in opening an issue about something when I can't even create a more minimal example of it.

Reply:
I suspect its related: https://github.com/rust-lang/rust/issues/122519

One way of handling this might be to use explicit paths (`::core` instead of `core`) but I'm not sure how to do that here as the macro should handle that. Another suggestion might be to add the `#[allow` into the `proc_macro` generated code, but I don't know where to add that best. So I can't really test if it's really related to the linked issue.